crates/hir/src/lib.rs

```diff
@@ -2220,13 +2219,17 @@ impl LifetimeParam {
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
 pub struct ConstParam {
-    pub(crate) id: ConstParamId,
+    pub(crate) id: TypeOrConstParamId,
 }
 
 impl ConstParam {
+    pub fn merge(self) -> TypeOrConstParam {
+        TypeOrConstParam { id: self.id }
+    }
+
     pub fn name(self, db: &dyn HirDatabase) -> Name {
         let params = db.generic_params(self.id.parent);
-        params.consts[self.id.local_id].name.clone()
+        params.types[self.id.local_id].name().unwrap().clone()
```

Comment:
This `unwrap` doesn't sit well with me(even though it should never happen), going with `Name::missing()` might be nicer in that case